sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts

```diff
@@ -748,6 +859,403 @@ export class TextAnalyticsClient {
+    tasks: JobManifestTasks,
+    language?: string,
+    options?: AnalyzeOptions
+  ): Promise<GeneratedClientAnalyzeResponse>;
+  /**
+   * Submit a collection of text documents for analysis. Specify one or more unique tasks to be executed.
+   * @param documents Collection of documents to analyze
+   * @param tasks Tasks to execute.
+   * @param options The options parameters.
+   */
+  public async analyze(
+    documents: TextDocumentInput[],
+    tasks: JobManifestTasks,
+    options?: AnalyzeOptions
+  ): Promise<GeneratedClientAnalyzeResponse>;
+  public async analyze(
+    documents: string[] | TextDocumentInput[],
+    tasks: JobManifestTasks,
+    languageOrOptions?: string | AnalyzeOptions,
+    options?: AnalyzeOptions
+  ): Promise<GeneratedClientAnalyzeResponse> {
+    let realOptions: RecognizeLinkedEntitiesOptions;
+    let realInputs: TextDocumentInput[];
+
+    if (!Array.isArray(documents) || documents.length === 0) {
+      throw new Error("'documents' must be a non-empty array");
+    }
+
+    if (isStringArray(documents)) {
+      const language = (languageOrOptions as string) || this.defaultLanguage;
+      realInputs = convertToTextDocumentInput(documents, language);
+      realOptions = options || {};
+    } else {
+      realInputs = documents;
+      realOptions = (languageOrOptions as RecognizeLinkedEntitiesOptions) || {};
+    }
+
+    const { span, updatedOptions: finalOptions } = createSpan(
+      "TextAnalyticsClient-analyze",
+      realOptions
+    );
+
+    try {
+      return await this.client.analyze({
+        body: {
+          analysisInput: {
+            documents: realInputs
+          },
+          tasks: addEncodingParamToAnalyzeInput(tasks)
+        },
+        ...operationOptionsToRequestOptionsBase(finalOptions)
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  private async healthInternal(
+    documents: TextDocumentInput[],
+    options?: BeginHealthInternalOptions
+  ): Promise<HealthResponse> {
+    const { span, updatedOptions: finalOptions } = createSpan(
+      "TextAnalyticsClient-health",
+      addStrEncodingParam(options)
+    );
+
+    try {
+      return await this.client.health(
+        { documents: documents },
+        operationOptionsToRequestOptionsBase(finalOptions)
+      );
+    } catch (e) {
+      let exception = handleInvalidDocumentBatch(e);
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: exception.message
+      });
+      throw exception;
+    } finally {
+      span.end();
+    }
+  }
+
+  private async _listHealthcareEntitiesByPage(
+    jobId: string,
+    input: TextDocumentInput[],
+    options: HealthStatusOptions = {}
+  ): Promise<PagedAsyncIterableHealthEntities> {
+    const iter = this._listHealthcareEntities(jobId, input, options);
+    const pagedIterator = {
```

Comment:
This looks great, but I can't help but wonder if this could be replaced by something simpler like

```typescript
const pagedIterator = Object.assign(this._listHealthcareEntities(jobId, input, options), {
  byPage: (settings?: PageSettings) => {
    const pageOptions = { ...options, top: settings?.maxPageSize };
    return this._listHealthcareEntitiesPaginated(jobId, input, pageOptions);
  });
```

since essentially all we're doing here is providing an escape to the underlying paged API and the unpaged iterator is just a convenience on top of that.

Reply:
nice!